shared/params.go

```diff
@@ -110,19 +110,28 @@ func ParseProduct(product string) (result Product, err error) {
 // ParseVersion parses the given version as a semantically versioned string.
 func ParseVersion(version string) (result *Version, err error) {
```

Comment:
Is the Version struct and ParseVersion need at all now? I was expecting them to disappear with this change, not grow :)

It looks like before this change they were only used to validate and throw an error in ParseProduct, the struct isn't used. After the change it also looks like this is the only purpose? Since these strings won't comply with semver and will have varying number of components, do we need to parse them at all? Just checking that they match a regex seems like less code, and clearer that only validation is going on.

Reply:
I intend to use them properly at a later date - this is validation + prepwork mixed together :)

Reply:
OK, will take another look when it's used for realz :)

(I'm guessing funny things will happen when the number of components is less than 4 or when some part isn't entirely numerical, but will see.)